### PR TITLE
refactor: network state and API client

### PR DESCRIPTION
**Pull request by He1DAr**

### CODE REVIEW

File: src/common/api/index.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export namespace API {
+  export function getClient() {
+    return true;
+  }
+}
```
Comment (kyranjamie) [inline src/common/api/index.ts:5]: Why do we need namespace? There's much talk of it being a deprecated feature https://michelenasti.com/2019/01/23/is-typescript-namespace-feature-deprecated.html

File: src/common/state/globalSlice.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      state.activeNetworkUrl = DEFAULT_MAINNET_SERVER;
+    },
+  },
+});
+
+export const { setActiveNetworkUrl, addCustomNetwork, removeCustomNetwork } = slice.actions;
+
+const selectGlobal = (state: RootState) => state.global;
+export const selectActiveNetworkUrl = () => useAppSelector(state => state.global.activeNetworkUrl);
+export const selectNetworks = () =>
```
Comment (kyranjamie) [inline src/common/state/globalSlice.ts:46]: Isn't this a hook, thus needs `use` prefix?

File: src/features/blocks-list/hooks.ts [hunk 3]
```diff
@@ -1,8 +1,32 @@
...
+export function useBlocksListOld(limit = DEFAULT_BLOCKS_LIST_LIMIT) {
   return useInfiniteQueryAtom<InfiniteData<BlocksListResponse> | undefined>(blocksListState(limit));
 }
+
+export function useBlocksList(limit = DEFAULT_BLOCKS_LIST_LIMIT) {
+  const { blocksApi } = useApi();
+  return useQuery({
+    queryKey: ['blocks/CONFIRMED', limit],
+    queryFn: async () => {
+      return (await blocksApi.getBlockList({
```
Comment (kyranjamie) [inline src/features/blocks-list/hooks.ts:25]: `await` keyword not needed when returning promise from async fn

File: src/common/api/index.ts (hunk 1 above)
Comment (kyranjamie) [reply src/common/api/index.ts:5]: or rather, not deprecated, but generally best avoided being usd

File: src/common/state/globalSlice.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      delete state.customNetworks[action.payload];
+      state.activeNetworkUrl = DEFAULT_MAINNET_SERVER;
+    },
+  },
+});
+
+export const { setActiveNetworkUrl, addCustomNetwork, removeCustomNetwork } = slice.actions;
+
+const selectGlobal = (state: RootState) => state.global;
+export const selectActiveNetworkUrl = () => useAppSelector(state => state.global.activeNetworkUrl);
```
Comment (kyranjamie) [inline src/common/state/globalSlice.ts:45]:
```suggestion
export const selectActiveNetworkUrl = createSelector(selectGlobal, state => state.activeNetworkUrl);
```

ideally compose using the other selectors

**Comment by He1DAr** [comment]
Will reopen as draft pr as it's not ready for review yet.

**Pull request closed without merging**